app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -596,12 +596,12 @@ public void onAnimationStart(Animator animator) {
 
             @Override
             public void onAnimationEnd(Animator animator) {
+                pokemonMarker.getMarker().remove();
+                markerList.remove(pokemonMarker);
             }
 
             @Override
             public void onAnimationCancel(Animator animator) {
-                pokemonMarker.getMarker().remove();
```

Comment:
Because onAnimationCancel() callback will never be called, except you call animator.cancel(). So I think in this case, we only need to handler in onAnimationEnd();